samples/tutorial/tutorial-final/src/main/java/workflow/tutorial/TodoModel.kt

```diff
@@ -1,6 +1,15 @@
 package workflow.tutorial
 
+import com.squareup.workflow1.ui.TextController
+import com.squareup.workflow1.ui.WorkflowUiExperimentalApi
+
+@OptIn(WorkflowUiExperimentalApi::class)
 data class TodoModel(
-  val title: String,
-  val note: String
-)
+  val title: TextController,
+  val note: TextController
+) {
+  constructor(
+    title: String,
+    note: String
+  ) : this(TextController(title), TextController(note))
+}
```

Comment:
is this used directly in the rendering?

Reply:
Yes. That's the normal pattern for `TextController`, which is why I've pushed back when anyone talks about generalizing it.